src/components/Todo.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+const Todo = ({text,todo, todos,setTodos}) => {
+    //Events
+    const deleteHandler = () => {
+        setTodos(todos.filter((el) => el.id !==todo.id))
+    };
+   
+    const completeHandler = () => {
+        setTodos(todos.map(item =>{
```

Comment:
How 'bout only checking `completed` since it's the only changing value:

```js
todos.map((item) => ({
  ...item,
  completed: item.id === todo.id
    ? !item.completed
    : item.completed
}))
```

Also prefer `todo` rather than `item`